studio/components/interfaces/Auth/Users/UserDropdown.tsx

```diff
@@ -1,149 +1,93 @@
+import { Button, Dropdown, IconMail, IconMoreHorizontal, IconShieldOff, IconTrash } from 'ui'
 
+import { useParams } from 'common'
+import { confirmAlert } from 'components/to-be-cleaned/ModalsDeprecated/ConfirmModal'
+import { useUserDeleteMFAFactorsMutation } from 'data/auth/user-delete-mfa-factors-mutation'
+import { useUserDeleteMutation } from 'data/auth/user-delete-mutation'
+import { useUserResetPasswordMutation } from 'data/auth/user-reset-password-mutation'
+import { useUserSendMagicLinkMutation } from 'data/auth/user-send-magic-link-mutation'
+import { useUserSendOTPMutation } from 'data/auth/user-send-otp-mutation'
 import { useStore } from 'hooks'
 import { timeout } from 'lib/helpers'
-import { post, delete_ } from 'lib/common/fetch'
-import { API_URL } from 'lib/constants'
 import { PageContext } from 'pages/project/[ref]/auth/users'
-import { confirmAlert } from 'components/to-be-cleaned/ModalsDeprecated/ConfirmModal'
 import { User } from './Users.types'
 
-interface Props {
+interface UserDropdownProps {
   user: User
   canRemoveUser: boolean
   canRemoveMFAFactors: boolean
 }
 
-const UserDropdown: FC<Props> = ({ user, canRemoveUser, canRemoveMFAFactors }) => {
-  const PageState: any = useContext(PageContext)
+const UserDropdown = ({ user, canRemoveUser, canRemoveMFAFactors }: UserDropdownProps) => {
   const { ui } = useStore()
-  const [loading, setLoading] = useState<boolean>(false)
+  const { ref } = useParams()
+  const PageState: any = useContext(PageContext)
 
-  async function handleResetPassword() {
-    try {
-      setLoading(true)
-      const response = await post(`${API_URL}/auth/${PageState.projectRef}/recover`, user)
-      if (response.error) {
-        ui.setNotification({
-          category: 'error',
-          message: `Failed to send password recovery: ${response.error.message}`,
-        })
-      } else {
-        ui.setNotification({
-          category: 'success',
-          message: `Sent password recovery to ${user.email}`,
-        })
-      }
-    } catch (error: any) {
-      ui.setNotification({
-        category: 'error',
-        message: `Send password recovery failed: ${error?.message}`,
-      })
-    } finally {
-      setLoading(false)
-    }
+  const { mutateAsync: resetPassword, isLoading: isResetting } = useUserResetPasswordMutation()
+  const { mutateAsync: sendMagicLink, isLoading: isSendingLink } = useUserSendMagicLinkMutation()
+  const { mutateAsync: sendOTP, isLoading: isSendingOTP } = useUserSendOTPMutation()
+  const { mutateAsync: deleteUser, isLoading: isDeleting } = useUserDeleteMutation()
+  const { mutateAsync: deleteUserMFAFactors, isLoading: isDeletingFactors } =
+    useUserDeleteMFAFactorsMutation()
+  const isLoading = isResetting || isSendingLink || isSendingOTP || isDeleting || isDeletingFactors
+
+  const handleResetPassword = async () => {
+    if (!ref) return console.error('Project ref is required')
+    await resetPassword({ projectRef: ref, user })
```

Comment:
missing try/catch

Reply:
oh wait i can convert this to mutate, instead of mutateAsync